drivers/sensor/stm32_vbat/Kconfig

```
@@ -0,0 +1,10 @@
+# STM32 battery sensor configuration options
+
+# Copyright (c) 2022 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+config STM32_VBAT
+	bool "STM32 Vbat Sensor"
+	depends on (ADC && SOC_FAMILY_STM32)
```

```suggestion
	depends on (SOC_FAMILY_STM32 && !SOC_SERIES_STM32F1X)
	select ADC
```
Then:
- update description 
- remove `CONFIG_ADC=y` in sample

---

Pb : this creates a "Dependency loop"